devel/nlohmann-json/Portfile

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem                      1.0
+
+PortGroup                       github 1.0
+PortGroup                       cmake 1.1
+PortGroup                       cxx11 1.1
+
+github.setup                    nlohmann json 3.2.0 v
+name                            nlohmann-json
+categories                      devel
+platforms                       darwin
+supported_archs                 noarch
+license                         MIT
+maintainers                     {gmail.com:g.litenstein @Lord-Kamina} openmaintainer
+description                     Nlohmann JSON for modern C++.
+long_description                Nlohmann is a header-only (and optionally single-header) JSON library \
+                                written in vanilla C++11. Has Unicode support\; DOM and SAX parsers, among other features.
+checksums                       rmd160  3c04b206efdeaacf49f374b33eec5b86978eef6d \
+                                sha256  2131e11d0774005e68432c5c01ff0a595e6abf0ba7979738a2017ee8a69d4001 \
+                                size    114561095
+
+configure.pre_args-append       -DBUILD_TESTING=OFF -DJSON_BuildTests=OFF
+configure.pre_args-delete       -DCMAKE_POLICY_DEFAULT_CMP0025=NEW
```

Comment:
is there a reason to alter this default? It seems to work fine and pass all tests without this deletion.

Reply:
Not really. I must have gotten a warning about it at some point but now that you mention it, I'll remove that.

Reply:
`-DCMAKE_POLICY_DEFAULT_CMP0025=NEW` changes the compiler identification of Xcode clang from "Clang" to "AppleClang". There are a couple places in benchmarks/thirdparty/benchmark/CMakeLists.txt and benchmarks/thirdparty/benchmark/cmake/HandleGTest.cmake where `"${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"` is used to identify clang. If those cmake files are used in our build, those `STREQUAL`s would need to be changed to `MATCHES` for compatibility with CMP0025.

Reply:
What would you rather do in this case? Patch those files or ensure old behavior? (Which would anyway probably be better served by changing to old explicitly instead of deleting as I'm doing now)

Reply:
We would like to fix projects to be compatible with CMP0025. So we should fix the files and send the fixes upstream.

Reply:
I've submitted a PR to nlohmann-json, and they told me it's actually a google library so I submitted it over there also. For now, as you can see, I added the relevant patches to the port.